lib/buffer.js

```diff
@@ -933,8 +940,12 @@ Buffer.prototype.readUInt32BE = function(offset, noAssert) {
 
 
 Buffer.prototype.readIntLE = function(offset, byteLength, noAssert) {
+  if (!noAssert)
+    checkByteLength(byteLength);
```

Comment:
The `checkOffset` uses the `byteLength`, and assumes it is defined. Also, in the master branch, the `offset` and `byteLength` are modified before the checkOffset function is invoked.